tangelo/linq/gate.py

```diff
@@ -84,6 +89,18 @@ def check_qubit_indices(qubit_list, label):
         if len(all_involved_qubits) != len(set(all_involved_qubits)):
             raise ValueError(f"There are duplicate qubits in the target/control qubits")
 
+        # Check for the number of qubits. If it is a custom gate (not defined in
+        # the X_QUBIT_GATES sets), ignore the check. We also do not prevent
+        # the creation of multi-controlled gates (begins with "C").
+        if name.upper() in ONE_QUBIT_TARGET_GATES:
+            n_qubits = 1
+        elif name.upper() in TWO_QUBIT_TARGET_GATES:
+            n_qubits = 2
+        else:
+            n_qubits = len(target)
+        if len(target) != n_qubits:
+            raise ValueError(f"There are too many target/control qubits for this gate.")
```

Comment:
1. Replace "this gate" by the actual name of the gate, let's be useful there. I'd rather use `n_targets` than `n_qubits` as we use `n_qubits` many other places. We could say something like.

`f"Gate {g.name} : expected {n_qubits} target qubits, but got {target}"`

2. Your current code only seems to check that the number of targets is good, I'm not sure why your current error message says 'control' as well. Should there be a check for control, or none at all ?

Reply:
Oh this is a mistake as I changed my mind last minute on the control check.